File: billboard-client/src/main/java/com/example/billboardclient/MessageController.java
package com.example.billboardclient;

import com.netflix.hystrix.contrib.javanica.annotation.HystrixCommand;
import com.netflix.hystrix.contrib.javanica.annotation.HystrixProperty;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class MessageController {

    @Autowired
    private MessageService messageService;

    @GetMapping("/")
    public String getMessage(@RequestParam(value = "id", required = false) String billboardId)
    {
      return messageService.getMessage(billboardId);
    }

    @GetMapping("/slow")
    public String slow(){
        return messageService.slowOperation("test");
    }

}

File: dotnet/billboard-client/Controllers/MessageController.cs
using BillboardClient.Services;
using Microsoft.AspNetCore.Mvc;

namespace BillboardClient.Controllers
{
    public class MessageController : Controller
    {
        private readonly MessageService _messageService;

        public MessageController(MessageService messageService)
        {
            _messageService = messageService;
        }

        [Route("/")]
        public IActionResult GetMessage(string billboardId)
        {
            return Json(_messageService.GetMessage(billboardId));
        }

        [Route("/slow")]
        public IActionResult Slow() => Json(_messageService.SlowOperation("test"));
    }
}